site/blog/2022/deploy-static-website.mdx

```diff
@@ -0,0 +1,193 @@
+```sql TheButton[Trigger The CodeBuild Project]="Trigger The CodeBuild Project"
+SELECT * FROM start_build('deploy-static-website', 'us-east-1');
+```
+This will trigger the CodeBuild project to start. After a while we should be able to see the files appearing in our S3 bucket. We can access our React app using the endpoint for the S3 static website hosting. As we already mentioned, to get the endpoint we can use `get_bucket_website_endpoint` helper function.
+```sql TheButton[Get The Bucket Website Endpoint]="Get The Bucket Website Endpoint"
+SELECT get_bucket_website_endpoint('<bucket-name>');
+```
+By visiting the link returned by the above function, you can see our sample app has been deployed. The problem is that S3 static website hosting does not support HTTPS, and therefore we need to use a CloudFront distribution in order to have HTTPS connection.
+
+## Create a CloudFront Distribution
+Serving files in a bucket to the public using pure S3 isn't a good idea. In our case because the S3 static website hosting does not provide an HTTPS endpoint, but in most cases because the [data transfer rates for S3](https://aws.amazon.com/s3/pricing/) aren't cheap and can grow out of control. Also, the speed in which the users can access to the bucket objects will increase if you use a CDN because they'll be delivered to the users from the nearest edge server.
+
+With the above in mind, let's create a CloudFront distribution for our S3 bucket. But first, we need to install the `aws_cloudfront` module to be able to leverage its abilities.
+```sql TheButton[]=""
+SELECT iasql_install('aws_cloudfront');
+```
+Then create the distribution:
+```sql TheButton[Create a CloudFront Distribution]="Create a Distribution For Bucket Website"
+SELECT iasql_begin();
+INSERT INTO distribution (caller_reference, origins, default_cache_behavior)
+VALUES ('my-website', '[
+  {
+    "DomainName": "' || (SELECT get_bucket_website_endpoint('<bucket-name>')) || '",
+    "Id": "my-website-origin",
+    "CustomOriginConfig": {
+      "HTTPPort": 80,
+      "HTTPSPort": 443,
+      "OriginProtocolPolicy": "http-only"
+    }
+  }
+]', '{
+  "TargetOriginId": "my-website-origin",
+  "ViewerProtocolPolicy": "allow-all",
+  "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6"
+}');
+SELECT iasql_commit();
+```
+
+We can access our website through the CloudFront distribution domain name. To get it, we can simply run the following query:
+```sql TheButton[Get the Distribution Domain Name]="Get the Distribution Domain Name"
+SELECT domain_name FROM distribution WHERE caller_reference = 'my-website';
+```
+It supports HTTPS, it's fast, it's cheaper than directly serving on S3, and it can be easily connected to a Route53 domain.
```

Comment:
maybe add some reference on how to configure the domain to point to the cloudfront distribution?

Reply:
We can use an AlternateName in CloudFront to do this. But our CloudFront module does not yet support this. Maybe I can add support for that functionality as a follow-up PR and then add more details to the post?